modules/opengl/src/shader/shader.cpp

```diff
@@ -97,11 +106,11 @@ Shader::Shader(const Shader &rhs) : id_{glCreateProgram()}, warningLevel_{rhs.wa
     ShaderManager::getPtr()->registerShader(this);
 }
 
-Shader::Shader(Shader &&rhs) : id_{rhs.id_}, ready_(rhs.ready_), warningLevel_{rhs.warningLevel_} {
+Shader::Shader(Shader &&rhs) : ready_(rhs.ready_), warningLevel_{rhs.warningLevel_} {
 
     ShaderManager::getPtr()->unregisterShader(&rhs);
-
-    rhs.id_ = 0;
+    prog_ = rhs.prog_;
```

Comment:
What about `ShaderManager::getPtr()->unregisterShader(this);` ?

See destructor and move assignment.

Reply:
Not necessary in move constructor